packages/cf-component-select/src/Select.js

```diff
@@ -76,7 +148,143 @@ Select.propTypes = {
   multi: PropTypes.bool,
   onChange: PropTypes.func,
   async: PropTypes.bool,
-  creatable: PropTypes.bool
+  creatable: PropTypes.bool,
+  menuRenderer: PropTypes.func,
+  arrowRenderer: PropTypes.func,
+  clearRenderer: PropTypes.func,
+  isValidNewOption: PropTypes.func,
+  noResultsText: PropTypes.string,
+  promptTextCreator: PropTypes.func,
+  newOptionCreator: PropTypes.func,
+  menuTitle: PropTypes.string
+};
+
+const singleOrPlaceholder = theme => ({
+  bottom: 0,
+  color: theme.colorGrayDark,
+  left: 0,
+  lineHeight: '32px',
+  paddingLeft: '10px',
+  paddingRight: '10px',
+  position: 'absolute',
+  right: 0,
+  top: 0,
+  maxWidth: '100%',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap'
+});
+
+const styles = ({ theme, disabled }) => {
+  const values = {
+    boxSizing: 'border-box',
+    fontSize: '0.86667rem'
+  };
+
+  return {
+    ...values,
+    '& div': values,
+    '& input': values,
+    '& span': values,
+
+    flex: theme.flex,
+    margin: theme.margin,
+    width: theme.width,
+    display: theme.display,
+    verticalAlign: theme.verticalAlign,
+    position: 'relative',
+    opacity: disabled ? 0.5 : 1,
+
+    '& .Select-control': {
+      backgroundColor: disabled ? theme.colorGrayDark : theme.colorWhite,
+      border: `1px solid ${theme.color.hail}`,
+      borderRadius: theme.borderRadius,
+      color: theme.colorGrayDark,
```

Comment:
We're avoiding using the colors outside the colors object. https://cloudflare.github.io/cf-ui/#cf-design-colors should be the only colors you need. If you need a new color we can add it.

Reply:
Surely we should remove anything from the theme that shouldn't be used, or at least mark it deprecated?

Reply:
Yeah. I'm working on doing that. Been busy with migration things. But I'll get to our color stuff soon.

Reply:
Why were the added to theme to start with?

Reply:
I think they were added before we had the new styleguide stuff in place. I think this will change once the new new styleguide lands.